more `cd` case judgement for unexpected result

As running, I found a bug that will trigger the `if` part of the code incorrectly near [here](https://github.com/eahydra/socks/blob/master/socks4.go#L134-L136), and the following error would occur:

```
panic: runtime error: index out of range

goroutine xxx [running]:
panic(0x713020, 0xc04200c040)
	D:/Go/src/runtime/panic.go:500 +0x1af
github.com/eahydra/socks.(*Socks4Client).Dial(0xc042164bc0, 0x7675cb, 0x3, 0xc04224db00, 0x12, 0x0, 0x0, 0x0, 0x0)
	C:/Users/SuperFashi/GoProjects/src/github.com/eahydra/socks/socks4.go:136 +0x9f8
github.com/eahydra/socks.(*Socks4Client).Dial-fm(0x7675cb, 0x3, 0xc04224db00, 0x12, 0x4, 0xc04224db00, 0x12, 0x0)
```

A patch would be simple.

--- a/socks4.go
+++ b/socks4.go
@@ -131,7 +131,7 @@ func (s *Socks4Client) Dial(network, address string) (net.Conn, error) {
 	if buff[1] != socks4Granted {
 		cd := int(buff[1]) - socks4Granted
 		failure := "unknown error"
-		if cd < len(socks4Errors) {
+		if cd < len(socks4Errors) && cd >= 0 {
 			failure = socks4Errors[cd]
 		}
 		return nil, errors.New("socks: SOCKS4 server at " + s.address + " failed to connect: " + failure)

